### PR TITLE
Feature/#174 bulk delete

### PR DESCRIPTION
**Pull request by KarlSjostrand**
Deleting several images at once did not scale well and caused a large amount of internal events. This PR changes this such that deletes are happening all-at-once where possible in both database and storage backends. The associated events are now also in "bulk mode", where a single event is emitted after deleting an entire batch of images. The internal API is also simplified, relying more often on image IDs and less often on entire image objects which must be fetched from the database.

### CODE REVIEW

File: src/main/scala/se/nimsa/sbx/app/routing/ImageRoutes.scala [hunk 1]
```diff
@@ -59,58 +60,46 @@ trait ImageRoutes {
...
+          parameters((
+            'framenumber.as[Int] ? 1,
+            'windowmin.as[Int] ? 0,
+            'windowmax.as[Int] ? 0,
+            'imageheight.as[Int] ? 0)) { (frameNumber, min, max, height) =>
+            get {
+              onComplete(Future(storage.readPngImageData(imageId, frameNumber, min, max, height))) {
+                case Success(bytes) => complete(HttpEntity(`image/png`, bytes))
+                case Failure(e) if e.isInstanceOf[NotFoundException] => complete(NotFound)
+                case Failure(_)  => complete(NoContent)
```
Comment (karin-exini) [inline src/main/scala/se/nimsa/sbx/app/routing/ImageRoutes.scala]: Why is this failure completed with NoContent? It seems as though the exception is lost in space and time.

File: src/main/scala/se/nimsa/sbx/metadata/MetaDataDAO.scala [hunk 2]
```diff
@@ -530,72 +530,6 @@ class MetaDataDAO(val dbConf: DatabaseConfig[JdbcProfile])(implicit ec: Executio
 
   def updateImage(image: Image): Future[Int] = db.run(updateImageAction(image))
 
-  // *** Deletes ***
```
Comment (karin-exini) [inline src/main/scala/se/nimsa/sbx/metadata/MetaDataDAO.scala]: Where has all this gone?

File: src/main/scala/se/nimsa/sbx/app/routing/ImageRoutes.scala (hunk 1 above)
Comment (karin-exini) [inline src/main/scala/se/nimsa/sbx/app/routing/ImageRoutes.scala]: Ah, I think I see now, this corresponds to no images found.

File: src/main/scala/se/nimsa/sbx/metadata/MetaDataDAO.scala (hunk 2 above)
Comment (KarlSjostrand) [inline src/main/scala/se/nimsa/sbx/metadata/MetaDataDAO.scala]: Only one delete method (to rule them all), the bulk delete of images.

File: src/main/scala/se/nimsa/sbx/app/routing/ImageRoutes.scala (hunk 1 above)
Comment (KarlSjostrand) [inline src/main/scala/se/nimsa/sbx/app/routing/ImageRoutes.scala]: Thanks for pointing this out. Although we are picky with the SOP Classes and transfer syntaxes we support, some datasets cannot be rendered as PNG, and users can always allow more SOP Classes. But returning NoContent in those cases is too benign. I changed this to 501 NotImplemented and updated the API spec.

**Pull request merged**